### PR TITLE
[PodTargetInstaller] Symlink static lib custom modulemaps into the Headers directory

### PR DESCRIPTION
**Pull request by segiddins**
This means we wont be overwriting the underlying file, so `pod install` wont cause full recompiles

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 1]
```diff
@@ -644,7 +644,7 @@ def apply_xcconfig_file_ref_to_resource_bundle_targets(resource_bundle_targets,
           def create_module_map(native_target)
             return super(native_target) unless custom_module_map
 
-            path = target.module_map_path
+            path = Target.instance_method(:module_map_path).bind(target).call
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: can you explain why this change?

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 2]
```diff
@@ -654,6 +654,11 @@ def create_module_map(native_target)
               update_changed_file(generator, path)
               add_file_to_support_group(path)
 
+              if path != (linked_path = target.module_map_path)
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: can this be tested?

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 3]
```diff
@@ -776,6 +781,11 @@ def add_placeholder_target
           # @return [Void]
           #
           def add_swift_static_library_compatibility_header_phase(native_target)
+            if custom_module_map
+              raise Informative, 'Using swift static libraries with custom module maps does not yet work.' \
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: nit: "Swift"

**Comment by dnkoutso** [comment]
How were we overwriting the file? I thought `update_changed_file` was not changing the contents of files written.

**Comment by dnkoutso** [comment]
Overall LGTM!

**Comment by segiddins** [comment]
> How were we overwriting the file? I thought update_changed_file was not changing the contents of files written.

Because it was in `Pods/Headers`, it was always getting deleted in `Installer#clean_sandbox`

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb (hunk 1 above)
Comment (segiddins) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: sure, will add this in a comment or factor this to make it less hideous:

We want always to store the actual file in the `Target Support Files` directory in the sandbox, but the `PodTarget#module_map_path` method refers to the path used to _import_ the module map in build settings, which needs to be next to the headers in question

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb (hunk 2 above)
Comment (segiddins) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: integration specs cover this pretty thoroughly, I can look into adding a unit test

**Comment by segiddins** [comment]
If this looks good, I'll go ahead and update the submodule and write some unit tests

**Comment by dnkoutso** [comment]
👍 verified in a sample project that rebuilds do not happen.
